resteasy-core/src/main/java/org/jboss/resteasy/plugins/interceptors/GZIPEncodingInterceptor.java

```diff
@@ -61,46 +63,91 @@ public synchronized void commit()
             // GZIPOutputStream constructor writes to underlying OS causing headers to be written.
             // so we swap gzip OS in when we are ready to write.
             gzip  = new EndableGZIPOutputStream(delegate);
-            delegate = gzip;
+            delegate = new BlockingAsyncOutputStream(gzip);
          }
          catch (IOException e)
          {
             throw new RuntimeException(e);
          }
       }
+
+      synchronized void finish() throws IOException {
+         if (gzip != null) {
+            gzip.finish();
+         }
+      }
+
+      CompletionStage<Void> asyncFinish() {
+         try {
+            finish();
+         } catch (IOException e) {
+            CompletableFuture<Void> ret = new CompletableFuture<>();
+            ret.completeExceptionally(e);
+            return ret;
+         }
+         return CompletableFuture.completedFuture(null);
+      }

```

Comment:
Yeah, I agree. I don't think there's an easy way to do a fully async implementation, since the Java `GZIPOutputStream` implementation itself isn't async (see [here](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/zip/GZIPOutputStream.html#write(byte%5B%5D,int,int)))